extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php

```diff
@@ -53,80 +91,15 @@ private function getBaseUrl() {
 		return self::DISCUSSIONS_API_BASE_DEV;
 	}
 
-	private function apiRequest( $url ) {
-		$request = RequestContext::getMain()->getRequest();
-
-		$options = [
-			'returnInstance' => false,
-			'headers' => [
-				'cookie' => self::ACCESS_TOKEN_COOKIE_NAME . '=' .
-					$request->getCookie( self::ACCESS_TOKEN_COOKIE_NAME, '' ) . ';'
-			]
-		];
-
-		$data = Http::get( $url, 'default', $options );
-		$obj = json_decode( $data, true );
-		return $obj;
-	}
-
-	private function buildPost( $rawPost, $index ) {
-		global $wgContLang;
-
-		$timeAgo = wfTimeFormatAgo( wfTimestamp( TS_ISO_8601, $rawPost['creationDate']['epochSecond'] ) );
-		$userData = $rawPost['_embedded']['userData'][0];
-
-		return [
-			'author' => $rawPost['createdBy']['name'],
-			'authorAvatar' => $rawPost['createdBy']['avatarUrl'],
-			'commentCount' => $rawPost['postCount'],
-			'content' => $wgContLang->truncate( $rawPost['rawContent'], 120 ),
-			'createdAt' => $timeAgo,
-			'forumName' => wfMessage( 'embeddable-discussions-forum-name', $rawPost['forumName'] )->plain(),
-			'id' => $rawPost['id'],
-			'firstPostId' => $rawPost['firstPostId'],
-			'index' => $index,
-			'link' => '/d/p/' . $rawPost['id'],
-			'shareUrl' => $this->getBaseUrl() . '/d/p/' . $rawPost['id'],
-			'upvoteUrl' => $this->getUpvoteRequestUrl( $rawPost['firstPostId'] ),
-			'title' => $rawPost['title'],
-			'upvoteCount' => $rawPost['upvoteCount'],
-			'hasUpvoted' => $userData['hasUpvoted'],
-		];
-
-		return $post;
-	}
-
-	private function formatData( $rawData, $showLatest ) {
-		$rawThreads = $rawData['_embedded']['threads'];
+	public function getData( $showLatest, $limit, $category ) {
 		$sortKey = $showLatest ? self::SORT_LATEST_LINK : self::SORT_TRENDING_LINK;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)